dydx3/modules/private.py

```diff
@@ -707,47 +746,94 @@ def create_fast_withdrawal(
+        :param lp_stark_public_key: required
+        :type lp_stark_public_key: str
 
         :param client_id: optional
         :type client_id: str
 
+        :param expiration: optional
+        :type expiration: ISO str
+
+        :param expiration_epoch_seconds: optional
+        :type expiration_epoch_seconds: int
+
+        :param signature: optional
+        :type signature: str
+
         :returns: Transfer
 
         :raises: DydxAPIError
         '''
         client_id = client_id or random_client_id()
-        return self._post(
-            'fast-withdrawals',
-            {
-                'creditAsset': credit_asset,
-                'creditAmount': credit_amount,
-                'debitAmount': debit_amount,
-                # TODO: Signature verification should work regardless of case.
-                'toAddress': to_address.lower(),
-                'lpPositionId': lp_position_id,
-                'clientId': client_id,
-                'expiration': expiration,
-                'signature': signature,
-            },
+        if bool(expiration) == bool(expiration_epoch_seconds):
+            raise ValueError(
+                'Exactly one of expiration and expiration_epoch_seconds must '
+                'be specified',
+            )
+        expiration = expiration or epoch_seconds_to_iso(
+            expiration_epoch_seconds,
+        )
+        expiration_epoch_seconds = (
+            expiration_epoch_seconds or iso_to_epoch_seconds(expiration)
         )
 
+        if not signature:
+            if not self.stark_private_key:
+                raise Exception(
+                    'No signature provided and client was not' +
+                    'initialized with stark_private_key'
+                )
+            fact = get_transfer_erc20_fact(
+                recipient=to_address,
+                token_decimals=COLLATERAL_TOKEN_DECIMALS,
+                human_amount=credit_amount,
+                token_address=(
+                    TOKEN_CONTRACTS[COLLATERAL_ASSET][self.network_id]
+                ),
+                salt=nonce_from_client_id(client_id),
+            )
+            transfer_to_sign = SignableConditionalTransfer(
+                network_id=self.network_id,
+                sender_position_id=position_id,
+                receiver_position_id=lp_position_id,
+                receiver_public_key=lp_stark_public_key,
+                fact_registry_address=FACT_REGISTRY_CONTRACT[self.network_id],
+                fact=fact,
+                human_amount=debit_amount,
+                client_id=client_id,
+                expiration_epoch_seconds=expiration_epoch_seconds,
+            )
+            signature = transfer_to_sign.sign(self.stark_private_key)
+
+        params = {
+            'creditAsset': credit_asset,
+            'creditAmount': credit_amount,
+            'debitAmount': debit_amount,
+            # TODO: Signature verification should work regardless of case.
```

Comment:
Don't quite understand this TODO here or above, just wanted to make sure that it wasn't copied on accident

Reply:
Yeah this is a real TODO (on the backend, BAC-671)